### PR TITLE
grakn: init at version 1.8.4

### PR DESCRIPTION
**Pull request by haskie-lambda**
###### Motivation for this change
provide an easy way to use grakn.ai for independent projects

###### Things done
****

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

I was unaware that every commit should be of the specified form; but i guess that because it is an initial pull for a new package the commits can be squashed?

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -3393,6 +3393,12 @@
       fingerprint = "74B1 F67D 8E43 A94A 7554  0768 9CCC E364 02CB 49A6";
     }];
   };
+  haskie = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:3396]: Please put this into a separate commit with the message ``maintainers: add haskie``.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -168,6 +168,8 @@ in
   buildFHSUserEnvChroot = callPackage ../build-support/build-fhs-userenv { };
   buildFHSUserEnvBubblewrap = callPackage ../build-support/build-fhs-userenv-bubblewrap { };
 
+  grakn = callPackage ../servers/grakn/grakn.nix {};
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:171]:
```suggestion
  grakn = callPackage ../servers/grakn/grakn.nix { };
```

Please sort the package in the right section alphabetically.

File: pkgs/servers/grakn/grakn.nix [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+${graknHome}/grakn \$@; 
+            " > $out/grakn
+        
+        # make grakn executable
+        chmod +x $out/grakn
+        '';
+
+    doCheck = true;
+
+    meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/servers/grakn/grakn.nix:73]:
```suggestion
    meta = with lib; {
```

File: pkgs/servers/grakn/grakn.nix [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        graknFile=./${graknDir}/grakn
+
+        # replace the paths of the grakn boot script with the nix paths
+        sed -i "20s#^JAVA_BIN=java\$#JAVA_BIN=${openjdk}/bin/java#" $graknFile
+        sed -i "85s#java#${openjdk}/bin/java#" $graknFile
+        sed -i "54s#java#${openjdk}/bin/java#" $graknFile
+        sed -i "79s#java#${openjdk}/bin/java#" $graknFile
+        '';
+    installPhase = ''
+        # copy the files to $out
```
Comment (SuperSandro2000) [inline pkgs/servers/grakn/grakn.nix:47]:
```suggestion
```

File: pkgs/servers/grakn/grakn.nix [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+        # add a wrapper script to $out that will move grakn to $graknHome
+        # this is necessary because grakn needs a writable environment
+        echo "
+# on the first start copy everything to graknHome
+if [ ! -f ${graknHome}/grakn ]; then 
+    mkdir -p ${graknHome}; 
+    cp -r $out/${graknDir}/* ${graknHome}; 
+    
+    # correct permissions so that grakn and the user can write there 
+    # (by default they are readonly because of the cp
```
Comment (SuperSandro2000) [inline pkgs/servers/grakn/grakn.nix:60]:
```suggestion
```

File: pkgs/servers/grakn/grakn.nix [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    chmod -R u+rw ${graknHome}
+    chmod u+x ${graknHome}/grakn
+fi; 
+${graknHome}/grakn \$@; 
+            " > $out/grakn
+        
+        # make grakn executable
```
Comment (SuperSandro2000) [inline pkgs/servers/grakn/grakn.nix:67]:
```suggestion
```

File: pkgs/servers/grakn/grakn.nix [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+        chmod +x $out/grakn
+        '';
+
+    doCheck = true;
```
Comment (SuperSandro2000) [inline pkgs/servers/grakn/grakn.nix:71]: Please add a comment why we are not running tests.

File: pkgs/servers/grakn/grakn.nix [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+            then windowsSrc
+            else if stdenv.isDarwin 
+                then macSrc
+                else linuxSrc ;
+
+    buildDepends = [ openjdk ]; # grakn runs on java
+
+    buildPhase = ''
+        graknFile=./${graknDir}/grakn
+
```
Comment (SuperSandro2000) [inline pkgs/servers/grakn/grakn.nix:39]:
```suggestion
```

File: pkgs/servers/grakn/grakn.nix [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+        # replace the paths of the grakn boot script with the nix paths
+        sed -i "20s#^JAVA_BIN=java\$#JAVA_BIN=${openjdk}/bin/java#" $graknFile
+        sed -i "85s#java#${openjdk}/bin/java#" $graknFile
+        sed -i "54s#java#${openjdk}/bin/java#" $graknFile
+        sed -i "79s#java#${openjdk}/bin/java#" $graknFile
```
Comment (SuperSandro2000) [inline pkgs/servers/grakn/grakn.nix:44]:
```suggestion
        sed -i "20s#^JAVA_BIN=java\$#JAVA_BIN=${openjdk}/bin/java#" ${graknDir}/grakn
        sed -i "85s#java#${openjdk}/bin/java#" ${graknDir}/grakn
        sed -i "54s#java#${openjdk}/bin/java#" ${graknDir}/grakn
        sed -i "79s#java#${openjdk}/bin/java#" ${graknDir}/grakn
```

File: pkgs/servers/grakn/grakn.nix [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+    };
+
+
+    # picking the correct archive
+
+    src = if stdenv.hostPlatform.isWindows
+            then windowsSrc
+            else if stdenv.isDarwin 
+                then macSrc
+                else linuxSrc ;
```
Comment (SuperSandro2000) [inline pkgs/servers/grakn/grakn.nix:33]:
```suggestion
    src = if stdenv.hostPlatform.isWindows then windowsSrc
      else if stdenv.isDarwin then macSrc
      else linuxSrc ;
```

File: pkgs/servers/grakn/grakn.nix [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+
+    buildDepends = [ openjdk ]; # grakn runs on java
```
Comment (SuperSandro2000) [inline pkgs/servers/grakn/grakn.nix:35]:
```suggestion
    buildDepends = [ openjdk ];
```

File: pkgs/servers/grakn/grakn.nix [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+ 
+    macSrc = fetchzip {
+        url = "https://github.com/graknlabs/grakn/releases/download/2.0.0-alpha-9/grakn-core-all-mac-2.0.0-alpha-9.zip";
+        sha256 = "1w51anjd1gwsjqawjkd6yx4m0z5vd1cryap2i30ysm5jca8afn5j";
+    };
+
+
+    # picking the correct archive
+
+    src = if stdenv.hostPlatform.isWindows
```
Comment (SuperSandro2000) [inline pkgs/servers/grakn/grakn.nix:29]:
```suggestion

    src = if stdenv.hostPlatform.isWindows
```

File: pkgs/servers/grakn/grakn.nix [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+
+    windowsSrc = fetchzip {
+        url = "https://github.com/graknlabs/grakn/releases/download/2.0.0-alpha-9/grakn-core-all-windows-2.0.0-alpha-9.zip";
+        sha256 = "05gzpxwyscj63lqcl2nydfzqq7911a0wai2gm6hzqjqdadl9r363";
+    };
+ 
+    macSrc = fetchzip {
+        url = "https://github.com/graknlabs/grakn/releases/download/2.0.0-alpha-9/grakn-core-all-mac-2.0.0-alpha-9.zip";
+        sha256 = "1w51anjd1gwsjqawjkd6yx4m0z5vd1cryap2i30ysm5jca8afn5j";
+    };
```
Comment (SuperSandro2000) [inline pkgs/servers/grakn/grakn.nix:24]: Please move these into let in.

File: pkgs/servers/grakn/grakn.nix [hunk 14]
```diff
@@ -0,0 +1,85 @@
+{ stdenv, openjdk, fetchurl, graknHome?"~/.grakn_home", fetchzip}:
+
+let graknDir = "grakn-core-all-linux-2.0.0-alpha-9"; # name of the grakn download directory
+in
+stdenv.mkDerivation rec {
+    pname = "grakn";
+    version = "1.8.4";      # Grakn Version
+
+    # different archives for different machines
+
```
Comment (SuperSandro2000) [inline pkgs/servers/grakn/grakn.nix:10]:
```suggestion

```

File: pkgs/servers/grakn/grakn.nix [hunk 15]
```diff
@@ -0,0 +1,85 @@
+{ stdenv, openjdk, fetchurl, graknHome?"~/.grakn_home", fetchzip}:
+
+let graknDir = "grakn-core-all-linux-2.0.0-alpha-9"; # name of the grakn download directory
+in
+stdenv.mkDerivation rec {
+    pname = "grakn";
+    version = "1.8.4";      # Grakn Version
```
Comment (SuperSandro2000) [inline pkgs/servers/grakn/grakn.nix:7]:
```suggestion
    version = "1.8.4";
```

File: pkgs/servers/grakn/grakn.nix [hunk 16]
```diff
@@ -0,0 +1,85 @@
+{ stdenv, openjdk, fetchurl, graknHome?"~/.grakn_home", fetchzip}:
+
+let graknDir = "grakn-core-all-linux-2.0.0-alpha-9"; # name of the grakn download directory
```
Comment (SuperSandro2000) [inline pkgs/servers/grakn/grakn.nix:3]: Why has this directory 2.0.0.-alpha-9 in it when grakn version is 1.8.4?

File: pkgs/servers/grakn/grakn.nix [hunk 17]
```diff
@@ -0,0 +1,85 @@
+{ stdenv, openjdk, fetchurl, graknHome?"~/.grakn_home", fetchzip}:
```
Comment (SuperSandro2000) [inline pkgs/servers/grakn/grakn.nix:1]:
```suggestion
{ stdenv, openjdk, fetchurl, graknHome ? "~/.grakn_home", fetchzip}:
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please follow the contributing guide when naming your commits and reformat your code with two spaces indentation.

File: pkgs/servers/grakn/grakn.nix (hunk 7 above)
Comment (haskie-lambda) [reply pkgs/servers/grakn/grakn.nix:71]: where should i run which tests?
this is my first time contributing to nixpkgs; I tried to follow the contributing guideline and two different guides to create a package and I am not quite sure which tests you refer to here?

File: pkgs/servers/grakn/grakn.nix (hunk 8 above)
Comment (haskie-lambda) [reply pkgs/servers/grakn/grakn.nix:39]: why should this be removed?
i use this in lines 40,41,42,43 to avoid spelling mistakes

**Comment by haskie-lambda** [comment]
made commits compliant with contribution guidelines in 
#116392
